doc/_extensions/zephyr/link-roles.py

```diff
@@ -21,61 +23,74 @@
 
 def get_github_rev():
     try:
-        output = subprocess.check_output('git describe --exact-match',
-                                         shell=True, stderr=subprocess.DEVNULL)
+        output = subprocess.check_output(
+            "git describe --exact-match", shell=True, stderr=subprocess.DEVNULL
+        )
     except subprocess.CalledProcessError:
-        return 'main'
+        return "main"
 
-    return output.strip().decode('utf-8')
+    return output.strip().decode("utf-8")
 
 
 def setup(app):
-    rev = get_github_rev()
-
-    # Try to get the zephyr repository's GitHub URL from the manifest.
-    #
-    # This allows building the docs in downstream Zephyr-based
-    # software with forks of the zephyr repository, and getting
-    # :zephyr_file: / :zephyr_raw: output that links to the fork,
-    # instead of mainline zephyr.
-    baseurl = None
-    if west_manifest is not None:
-        try:
-            # This search tries to look up a project named 'zephyr'.
-            # If zephyr is the manifest repository, this raises
-            # ValueError, since there isn't any such project.
-            baseurl = west_manifest.get_projects(['zephyr'],
-                                                 allow_paths=False)[0].url
-            # Spot check that we have a non-empty URL.
-            assert baseurl
-        except ValueError:
-            pass
-
-    # If the search failed, fall back on the mainline URL.
-    if baseurl is None:
-        baseurl = 'https://github.com/zephyrproject-rtos/zephyr'
-
-    app.add_role('zephyr_file', autolink('{}/blob/{}/%s'.format(baseurl, rev)))
-    app.add_role('zephyr_raw', autolink('{}/raw/{}/%s'.format(baseurl, rev)))
+    baseurl = "https://github.com/zephyrproject-rtos/zephyr"
```

Comment:
can't baseurl for Zephyr be determined from the manifest?

Reply:
Not if the manifest does not exist, which is the case for the Zephyr docs. This is only used by the already existing `zephyr_file` and `zephyr_raw` roles

Reply:
Hmm Zephyr has a manifest: https://github.com/zephyrproject-rtos/zephyr/blob/main/west.yml

Reply:
I must have built it wrong locally, I will look into it